SecureAuth.Sdk/NumberProfileService.cs

```diff
@@ -0,0 +1,134 @@
+﻿using System;
+using System.Linq;
+
+namespace SecureAuth.Sdk
+{
+    public class NumberProfileService : INumberProfileService
+    {
+        private readonly ApiClient _apiClient;
+
+        protected internal NumberProfileService(ApiClient apiClient)
+        {
+            this._apiClient = apiClient;
+        }
+
+        /// <summary>
+        /// Evaluate the provided phone number for carrier status and details.
+        /// </summary>
+        /// <param name="userId">The unique user ID.</param>
+        /// <param name="phoneNumber">The phone number for evaluation.</param>
+        /// <returns>NumberProfileResponse</returns>
+        public NumberProfileResponse EvaluatePhoneNumber(string userId, string phoneNumber)
```

Comment:
Cool. Then can we make the parameterized EvaluatePhoneNumber method simply create the NumberProfileRequest object and call the EvaluatePhoneNumber(NumberProfileRequest) method? It's minor, but there is code duplication there that could be avoided.